test/generic_map_tests.jl

```diff
@@ -48,6 +48,12 @@ function generic_map_tests(mapf, inplace_mapf=nothing)
         inplace_mapf(x -> x*x, A, A)
         @test A == map(x -> x*x, Float64[1:10...])
 
+        # test inplace map working for non-array iterables
+        A = Float64[1:10...]
+        B = Tuple(1:10)
+        inplace_mapf(x -> x*x, A, B)
```

Comment:
Would be good to test the n-argument method as well? And maybe add a test for something that doesn't have `getindex`, like a generator `(i for i in 1:10)`.